tasks/src/main/scala/sbt/Execute.scala

```diff
@@ -153,7 +153,7 @@ private[sbt] final class Execute[F[_] <: AnyRef](
 
     results(node) = result
     state(node) = Done
-    progressState = progress.completed(progressState, node, result)
+    progress.completed(node, result)
```

Comment:
It's a little unfortunate that this is named completed rather than something like `onCompleted` because it isn't completely obvious that is being called for side effects only. But I know that this api has been around for a while.

Reply:
Since I am getting rid of `type S`, this is an opportunity to make breaking changes.

Reply:
```scala
trait ExecuteProgress[F[_]] {
  def initial(): Unit

  /**
   * Notifies that a `task` has been registered in the system for execution.
   * The dependencies of `task` are `allDeps` and the subset of those dependencies that
   * have not completed are `pendingDeps`.
   */
  def afterRegistered(task: F[_], allDeps: Iterable[F[_]], pendingDeps: Iterable[F[_]]): Unit

  /**
   * Notifies that all of the dependencies of `task` have completed and `task` is therefore
   * ready to run.  The task has not been scheduled on a thread yet.
   */
  def afterReady(task: F[_]): Unit

  /**
   * Notifies that the work for `task` is starting after this call returns.
   * This is called from the thread the task executes on, unlike most other methods in this callback.
   * It is called immediately before the task's work starts with minimal intervening executor overhead.
   */
  def beforeWork(task: F[_]): Unit

  /**
   * Notifies that the work for `task` work has finished.  The task may have computed the next task to
   * run, in which case `result` contains that next task wrapped in Left.  If the task produced a value
   * or terminated abnormally, `result` provides that outcome wrapped in Right.  The ultimate result of
   * a task is provided to the `completed` method.
   * This is called from the thread the task executes on, unlike most other methods in this callback.
   * It is immediately called after the task's work is complete with minimal intervening executor overhead.
   */
  def afterWork[A](task: F[A], result: Either[F[A], Result[A]]): Unit

  /**
   * Notifies that `task` has completed.
   * The task's work is done with a final `result`.
   * Any tasks called by `task` have completed.
   */
  def afterCompleted[A](task: F[A], result: Result[A]): Unit

  /** All tasks have completed with the final `results` provided. */
  def afterAllCompleted(results: RMap[F, Result]): Unit

  /** Notifies that either all tasks have finished or cancelled. */
  def stop(): Unit
}
```

Reply:
Nice. I like this.